### ISSUE TITLE
YAML Anchor / Fragments Support

### ISSUE THREAD

**Issue by adaliszk** [zavoloklom/docker-compose-linter#39]
## Version

v1.0.6

## YAML file where the error occurs

```yaml
name: fragment-example

x-shared-links: &shared-links
  links:
    - "foo:foo.example.svc.cluster.local"
    - "bar:bar.example.svc.cluster.local"

services:
  foo:
    image: traefik/whoami:v1.10

  bar:
    image: traefik/whoami:v1.10
    
  app_foo:
    image: traefik/whoami:v1.10
    <<: *shared-links

  app_bar:
    image: traefik/whoami:v1.10
    <<: *shared-links
```

## Command used to run

```bash
npx dclint .
```

## Additional context

- https://docs.docker.com/reference/compose-file/fragments
- https://yaml.org/spec/1.2.2/#3222-anchors-and-aliases

## Current behavior (console output)

```
ComposeValidationError: instancePath="/services/foo", schemaPath="#/additionalProperties", message="Validation error: must NOT have additional properties".  invalid-schema
```

## Expected behavior

The Anchors / Fragments should be resolved before validating the schema, and thus the error would not tell that additional properties are not allowed, as there are no actual keys like `<<` being used after parsing the YAML.

**Comment by zavoloklom** [comment]
@adaliszk Hi! I fixed it in v1.0.7, but there are some limitations - https://github.com/zavoloklom/docker-compose-linter?tab=readme-ov-file#anchor-handling

Please let me know if something goes wrong.

**Comment by adaliszk** [comment]
Thank you very much!
It looks promising; I will try to make time next week to add the tool as a listing step in my current project!
